flycheck.el

```diff
@@ -5440,9 +5441,10 @@ ERR is in BUFFER-FILES, replace it with the return value of the
 function `buffer-file-name'."
   (flycheck-error-with-buffer err
     (-when-let (filename (flycheck-error-filename err))
-      (when (seq-some (apply-partially #'flycheck-same-files-p
-                                       (expand-file-name filename cwd))
-                      buffer-files)
+      (when (or (seq-some (apply-partially #'flycheck-same-files-p
+                                           (expand-file-name filename cwd))
+                          buffer-files)
+                (not buffer-files))
```

Comment:
@joedevivo I'm sorry but that's not the solution: By adding this condition you're rewriting file names of _all_ errors of syntax checkers that didn't create any temporary files.  That's going to make a mess of many other checkers.

Please remove this change again!